### ISSUE TITLE
Return index of invalid objects

### ISSUE THREAD

**Issue by netcriptus** [schematics/schematics#307]
So, in my models I define a `user`, and a `user` has many `addresses`, which are defined in their own class. So when mapping the model to a schematics class, it becomes something like this

```
class UserResource(BaseSchema):

    """User resource."""

    __entity__ = User
    user_addresses = ListType(ModelType(UserAddressResource))
```

Here's the problem: imagine I have a user with 5 addresses, but 2 of them are invalid. Currently, schematics returns something like `UserAddresses: [{'street': ['Not a well formed street name.']}, {'street': ['Not a well formed street name.']}]`.

It would be great to have the index of those invalid objects in the list. My only current option is to run the validation again on each object until I find the invalid ones. But this is not practical, considering it will be doing the same work twice, and models usually have more than one relationship, so it's linked to many other models.

**Comment by di** [comment]
Seems like a duplicate of https://github.com/schematics/schematics/issues/242.

**Comment by netcriptus** [comment]
I think they are really close, but not duplicated. In that case, it was a list of values, and from there on schematics returns which values are invalid. In this case it is a list of objects, with its own validations.

So, in an object with 5 addresses, it is not ideal just to know that `street` is an invalid field, I need to know which of the 5 addresses has the invalid field.

**Comment by bintoro** [comment]
This indeed looks like a dupe of #242. It makes no difference whether the `ListType` contains integers or models, the goal is to get the index of the faulty item within the list.

**Comment by bintoro** [comment]
The proposed solution is described in #369 and implemented in #374, if anyone would like to comment on the new error message structure.
